### PR TITLE
RefineTetMeshLocally

### PR DESCRIPTION
**Pull request by moritzdannhauer**
This is a new module that is not related to EAB activities.

### CODE REVIEW

**Comment by dcwhite** [comment]
Testing/merging of this will wait until after EAB then.

**Comment by moritzdannhauer** [comment]
Sure, I do not know why travis fails though, builds fine on my machine ...

File: src/Core/Algorithms/Field/RefineTetMeshLocallyAlgorithm.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#define CORE_ALGORITHMS_FIELD_RefineTetMeshLocally_H 1
+
+// Datatypes that the algorithm uses
+#include <Core/Datatypes/DatatypeFwd.h> 
+
+// Base class for algorithm
+#include <Core/Algorithms/Base/AlgorithmBase.h>
+
+// for Windows support
+#include <Core/Algorithms/Field/share.h>
```
Comment (dcwhite) [inline src/Core/Algorithms/Field/RefineTetMeshLocallyAlgorithm.h]: @moritzdannhauer Make sure share.h is the final include in the file. The Windows build is broken due to this.

File: src/Modules/Fields/RefineTetMeshLocally.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+   DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef MODULES_FIELDS_RefineTetMeshLocally_H__
+#define MODULES_FIELDS_RefineTetMeshLocally_H__
+
+//#include <Core/Algorithms/Base/AlgorithmBase.h>
+#include <Dataflow/Network/Module.h>
+#include <Modules/Legacy/Fields/share.h>
```
Comment (dcwhite) [inline src/Modules/Fields/RefineTetMeshLocally.h]: @moritzdannhauer Also, wrong share.h include here: should be `Module/Fields/share.h`

**Comment by dcwhite** [comment]
@moritzdannhauer And please provide a test network at least.

**Comment by dcwhite** [comment]
Unit tests fail on my machine:
```
$ ./Algorithms_Field_Tests.exe --gtest_filter=*RefineTet*
Running main() from gtest_main.cc
Note: Google Test filter = *RefineTet*
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from RefineTetMeshLocallyAlgoTests
[ RUN      ] RefineTetMeshLocallyAlgoTests.Test59basicCutCases
Error: First module input (Field) is empty.
unknown file: error: SEH exception with code 0xc0000005 thrown in the test body.

[  FAILED  ] RefineTetMeshLocallyAlgoTests.Test59basicCutCases (0 ms)
[ RUN      ] RefineTetMeshLocallyAlgoTests.Test5MoreTheoreticalCutCases
unknown file: error: SEH exception with code 0xc0000005 thrown in the test body.

[  FAILED  ] RefineTetMeshLocallyAlgoTests.Test5MoreTheoreticalCutCases (0 ms)
[----------] 2 tests from RefineTetMeshLocallyAlgoTests (0 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (15 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] RefineTetMeshLocallyAlgoTests.Test59basicCutCases
[  FAILED  ] RefineTetMeshLocallyAlgoTests.Test5MoreTheoreticalCutCases

 2 FAILED TESTS
```

**Comment by dcwhite** [comment]
Test network seems OK though.

**Comment by dcwhite** [comment]
After pulling latest SCIRunTestData, unit tests pass.

**Pull request merged**